File: rb/features.rb
# KanjiData SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module KanjiDataFeatures
  def self.make_feature(name)
    case name
    when "base"
      KanjiDataBaseFeature.new
    when "ratelimit"
      KanjiDataRatelimitFeature.new
    when "retry"
      KanjiDataRetryFeature.new
    when "test"
      KanjiDataTestFeature.new
    when "timeout"
      KanjiDataTimeoutFeature.new
    else
      KanjiDataBaseFeature.new
    end
  end
end

File: py/kanjidata_sdk/features.py
# KanjiData SDK feature factory

from kanjidata_sdk.feature.base_feature import KanjiDataBaseFeature
from kanjidata_sdk.feature.test_feature import KanjiDataTestFeature


def _make_feature(name):
    features = {
        "base": lambda: KanjiDataBaseFeature(),
        "test": lambda: KanjiDataTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
